src/server/removal_buffer.rs

```diff
@@ -87,12 +87,11 @@ impl RemovalReader<'_, '_> {
     fn read(&mut self) -> impl Iterator<Item = (&Entity, &HashSet<ComponentId>)> {
         self.clear();
 
-        // TODO: Ask Bevy to provide an iterator over `RemovedComponentEvents`.
-        for &component_id in &self.components.0 {
-            let Some(component_events) = self.remove_events.get(component_id) else {
-                continue;
-            };
-
+        for (&component_id, component_events) in self
+            .remove_events
+            .iter()
+            .filter(|(component_id, _)| self.components.contains(*component_id))
```

Comment:
`remove_events` is `RemovedComponentEvents` with all removals. I need to filter them to iterate only on replicated components.
I can skip this filter, it will continue to work, but I will allocate memory for each removed component. This is why I decided to keep this early-return check.